questions/serializers.py

```diff
@@ -389,8 +390,14 @@ class Meta:
             "probability_yes_per_category",
             "percentiles",
             "slider_values",
+            "source",
         )
 
+    def validate_source(self, source):
+        if source not in [None, Forecast.SourceChoices.UI, Forecast.SourceChoices.API]:
+            raise serializers.ValidationError("Invalid source")
+        return source
```

Comment:
You can add `choices` to `source =serializers.CharField`  and rely on django auto-validation.